func-app/graphrag/index/cli.py

```diff
@@ -170,42 +151,110 @@ async def execute():
                 progress_reporter.info(str(output.result))
 
         if platform.system() == "Windows":
-            logging.info("All set to execute the workflows on Windows")
             import nest_asyncio  # type: ignore Ignoring because out of windows this will cause an error
 
             nest_asyncio.apply()
             loop = asyncio.get_event_loop()
             loop.run_until_complete(execute())
         elif sys.version_info >= (3, 11):
-            logging.info("Step6")
             import uvloop  # type: ignore Ignoring because on windows this will cause an error
 
             with asyncio.Runner(loop_factory=uvloop.new_event_loop) as runner:  # type: ignore Ignoring because minor versions this will throw an error
                 runner.run(execute())
         else:
-            logging.info("Step 6")
             import uvloop  # type: ignore Ignoring because on windows this will cause an error
 
             uvloop.install()
             asyncio.run(execute())
 
-    _run_workflow_async()
-    progress_reporter.stop()
-    if encountered_errors:
-        progress_reporter.error(
-            "Errors occurred during the pipeline run, see logs for more details."
-        )
+    ################ CONTEXT SWITCHING
+    if context_id:
+        if not is_valid_guid(context_id):
+            ValueError("ContextId is invalid: It should be a valid Guid")
+        if (context_operation != ContextSwitchType.Activate and context_operation != ContextSwitchType.Deactivate):
+            ValueError("ContextOperation is invalid: It should be Active or DeActive")
+        _switch_context(
+                root=root,
+                config=config,
+                reporter=progress_reporter,
+                context_operation=context_operation,
+                context_id=context_id,
+                community_level=community_level,
+                optimized_search=optimized_search,
+                use_kusto_community_reports=use_kusto_community_reports,
+            )
+        return
+
+    ################ INDEXING ITERATION
+    orig_storage_base=pipeline_config.storage.base_dir
+    orig_input_base="< RELATIVE PATH FROM func-app TO INPUT DIR>"
```

Comment:
Take a look at latest code and we need to incline it to read inputs Input_base_dir parameter that I added